src/components/CommandPalette.tsx

```diff
@@ -437,11 +438,18 @@ const CommandPalette: FC = () => {
             })}
           >
             {shortcuts.map(shortcut => {
-              // Check if the current gesture sequence ends with help gesture
               const helpCommand = commandById('help')
               const isHelpMatch =
                 shortcut.id === 'help' && (gestureInProgress as string)?.toString().endsWith(gestureString(helpCommand))
 
+              // Check if current gesture matches any command's gesture
+              const hasMatchingCommand = shortcuts.some(cmd => gestureInProgress === cmd.gesture)
```

Comment:
Calling `shortcuts.some` inside `shortcuts.map` is going to result in an O(n^2) running time. You should move this check outside `map` for efficiency.

Alternatively, you might be able to use `!selectedShortcut && !isHelpMatch`. That should correspond to the cancel gesture.